### DISCUSSION TITLE
Rotate VABS input file

### DISCUSSION THREAD

**Discussion by rhig2000** [wenbinyugroup/.github#26] [VABS Related Discussions] [answered]
I have VABS input files, but not the source data that generated the input files. I need to rotate the cross sections prior to running them through VABS. (This is due to the CAMRAD axis system issue where aero twist must be applied.) Is there an existing utility to facilitate manipulating a VABS input file?

**Comment by duhd1993** [comment]
https://github.com/wenbinyugroup/sgio/ is what you are looking for.

**Reply by rhig2000** [reply]
Thanks, duhd. From what I can tell, sgio converts various mesh formats such as Abaqus to VABS. But I'm not seeing where user can specify a twist angle using sgio.

**Reply by unitasium** [reply]
Currently sgio only provides functions to help users to read and write VABS input files. If you want to rotate the cross-section, you need to write your own function to transform nodal coordinates and element local frame one-by-one.

This seems a good feature that we can add it to sgio later.

**Comment by rhig2000** [comment]
Ok, thanks! I'll look into sgio.

Also, per documentation:
> To model initially curved and twisted beams, curve flag is 1, and three real numbers for the
> twist (k1) and curvatures (k2 and k3) should be provided in the very next line.

Is k1 in degrees or radians?
I specified k1 but echo returned the following error:
> Memory error: allocate E1V0                    

Any suggestions?

**Reply by wenbinyu-PU** [reply, marked as answer]
The unit of k1 is radian/length. Regarding the error message, please share with input file so that I can take a look at it.

**Reply by rhig2000** [reply]
Ok, so twist is twist rate. How does it affect the result?

**Reply by wenbinyu-PU** [reply]
twist will not affect the cross-sectional properties. k2 and k3 will affect the properties slightly.

**Reply by rhig2000** [reply]
Ok. So what is the purpose of the feature then? For what kind of application is it intended to be used? I was expecting it to be applicable for blade twist. We are using both CAMRAD and FLIGHTLAB but we'll need to maintain (2) sets of VABS files since they handle blade twist differently.

**Reply by wenbinyu-PU** [reply]
No particular use for cross-sectional analysis. However, it is needed in the 1D beam analysis.
